hs_composite_resource/tests/test_composite_resource.py

```diff
@@ -1370,3 +1395,43 @@ def test_unzip_aggregation(self):
         self.assertEqual(self.composite_resource.files.count(), 2)
         # there should be a referenced time series now
         self.assertEqual(RefTimeseriesLogicalFile.objects.all().count(), 1)
+
+    def test_unzip_aggregation_overwrite(self):
+        """Test that when a zip file gets unzipped at data/contents/ where the contents includes a
+        single file aggregation.  Testing aggregation is overwritten when overwrite is True"""
+
+        self.create_composite_resource()
+
+        self.add_file_to_resource(file_to_add=self.zipped_aggregation_file)
+        self.assertEqual(self.composite_resource.files.count(), 1)
+        self.assertEqual(RefTimeseriesLogicalFile.objects.all().count(), 0)
+        # unzip the above zip file  which should add one more file to the resource
+        zip_file_rel_path = os.path.join('data', 'contents', self.zipped_aggregation_file_name)
+        unzip_file(self.user, self.composite_resource.short_id, zip_file_rel_path,
+                   bool_remove_original=False, overwrite=True)
+
+        # resource should have 2 files now
+        self.assertEqual(self.composite_resource.files.count(), 2)
+        # there should be a referenced time series now
+        self.assertEqual(RefTimeseriesLogicalFile.objects.all().count(), 1)
+        ref_time = RefTimeseriesLogicalFile.objects.first()
+        ref_time.metadata.abstract = "overwritten"
+        ref_time.metadata.save()
+
+        self.assertEqual(RefTimeseriesLogicalFile.objects.first().metadata.abstract, "overwritten")
+
+        unzip_file(self.user, self.composite_resource.short_id, zip_file_rel_path,
+                   bool_remove_original=False, overwrite=True)
+
+        # resource should still have 2 files
+        self.assertEqual(self.composite_resource.files.count(), 2)
+        # there should still be a referenced time series
+        self.assertEqual(RefTimeseriesLogicalFile.objects.all().count(), 1)
+
+        self.assertEqual(RefTimeseriesLogicalFile.objects.first().metadata.abstract,
+                         "Discharge, cubic feet per second,Blue-green algae (cyanobacteria), "
+                         "phycocyanin data collected from 2016-04-06 to 2017-02-09 created on "
+                         "Thu Apr 06 2017 09:15:56 GMT-0600 (Mountain Daylight Time) from the "
+                         "following site(s): HOBBLE CREEK AT 1650 WEST AT SPRINGVILLE, UTAH, and "
+                         "Provo River at Charleston Advanced Aquatic. Data created by CUAHSI "
+                         "HydroClient: http://data.cuahsi.org/#.")
```

Comment:
@sblack-usu Instead of using assertEqual to verify the ref time series aggregation was overwritten, would it be sufficient just to assert abstract is not equal to the old one, i.e., ```overwritten```? That will simplify the test, and potentially more adaptable if the test file got changed for example.

Reply:
done